syntax = "proto3";

package bookstore;

service BookStore {
	rpc SayHello (Empty) returns (HellowReply) {}
	rpc GetAllBooks (Empty) returns (stream Book) {}
	rpc PostBook (Book) returns (Reply) {}
}

message Empty {
		
}

message HellowReply {
	string message = 1;
}

message Book {
	int32 id = 1;
	string name = 2;
	string author = 3;
}

message Reply {
	string msg = 1;
}
